toolchain/sem_ir/ids.cpp

```diff
@@ -86,24 +87,40 @@ auto NameId::ForIdentifier(IdentifierId id) -> NameId {
   }
 }
 
+auto NameId::ForPackageName(PackageNameId id) -> NameId {
+  if (auto identifier_id = id.AsIdentifierId(); identifier_id.has_value()) {
+    return ForIdentifier(identifier_id);
+  } else if (id == PackageNameId::Core) {
+    return NameId::Core;
+  } else if (!id.has_value()) {
+    return NameId::None;
+  } else {
+    CARBON_FATAL("Unexpected package ID {0}", id);
+  }
+}
+
 auto NameId::Print(llvm::raw_ostream& out) const -> void {
   if (!has_value() || index >= 0) {
     IdBase::Print(out);
     return;
   }
   out << Label << "(";
-  if (*this == SelfValue) {
-    out << "SelfValue";
-  } else if (*this == SelfType) {
-    out << "SelfType";
-  } else if (*this == PeriodSelf) {
-    out << "PeriodSelf";
-  } else if (*this == ReturnSlot) {
-    out << "ReturnSlot";
+  if (*this == Base) {
+    out << "Base";
+  } else if (*this == Core) {
+    out << "Core";
   } else if (*this == PackageNamespace) {
     out << "PackageNamespace";
-  } else if (*this == Base) {
-    out << "Base";
+  } else if (*this == ReturnSlot) {
```

Comment:
Looks like some of these got reordered, but not sure how/why? They're /almost/ alphabetically sorted - could/should they be entirely sorted? (just `PeriodSelf` is out of order, I think?)

Reply:
It was supposed to be the same order as in the .h file, which I'd reordered to alphabetical order (with `None` first). But I got it wrong here. Thanks, fixed.